src/runtime.rs

```diff
@@ -249,87 +250,87 @@ impl Runtime {
             database_key_index
         );
 
-        let mut from_stack = local_state.take_query_stack();
-        let from_id = std::thread::current().id();
-
-        // Make a "dummy stack frame". As we iterate through the cycle, we will collect the
-        // inputs from each participant. Then, if we are participating in cycle recovery, we
-        // will propagate those results to all participants.
-        let mut cycle_query = ActiveQuery::new(database_key_index);
-
-        // Identify the cycle participants:
-        let cycle = {
-            let mut v = vec![];
-            dg.for_each_cycle_participant(
-                from_id,
-                &mut from_stack,
-                database_key_index,
-                to_id,
-                |aqs| {
-                    aqs.iter_mut().for_each(|aq| {
-                        cycle_query.add_from(aq);
-                        v.push(aq.database_key_index);
+        let (me_recovered, others_recovered, cycle) = local_state.with_query_stack(|from_stack| {
```

Comment:
The diff here is just the closure wrapping causing an extra level of indentation